### ISSUE TITLE
TypeScript error in Vue files within a pnpm managed project

### ISSUE THREAD

**Issue by BearingMe** [nuxt/eslint#484]
### Environment

- Operating System: Darwin
- Node Version:     v20.16.0
- Nuxt Version:     3.12.4
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.7.1
- Builder:          -
- User Config:      compatibilityDate, devtools, modules
- Runtime Modules:  @nuxt/eslint@0.5.1
- Build Modules:    -

### Package

@nuxt/eslint-module

### Reproduction

- `pnpm dlx nuxi@latest init withpnpm`
  - pnpm
  - no git
- `cd withpnpm/`
- `npx nuxi module add eslint`
- `pnpm run dev`
-  add a interface to app.vue
```sh
echo '<script setup lang="ts">
interface ABC {
  a: string;
  b: number;
  c: boolean;
}
</script>

<template>
  <div>
    <NuxtRouteAnnouncer />
    <NuxtWelcome />
  </div>
</template>' > app.vue
```
- `npx eslint .` 

### Describe the bug

ESLint should raise a "\<interface> is defined but never used" error, however it is throwing "parsing error [...] keyword interface is reserved".

![image](https://github.com/user-attachments/assets/c8639b49-bbe3-4de0-8132-3c21d3bdf261)
 

### Additional context

After erasing `/node_modules`, `pnpm-lock.yaml` and creating a `.npmrc` file it now seems fixed. 

```
# .npmrc
shamefully-hoist=true
node-linker=hoisted
```

**Comment by antfu** [comment]
Did you have `typescript` installed in your project?

**Comment by antfu** [comment]
We temporarily closed this due to the lack of enough information. We could not identify whether it was a bug or a userland misconfiguration with the given info.
Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) to reopen the issue.
Thanks.

> [Why reproduction is required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by allan-rcos** [comment]
I'm with the same issue, have TypeScript installed. I installed direct from nuxt init:

<img width="551" height="177" alt="Image" src="https://github.com/user-attachments/assets/d1e05bd5-7141-43fc-a64c-55c007299620" />

**Comment by allan-rcos** [comment]
If somebody is having this issue in WebStorm, remember to change the IDE eslint config to Manual and configure the local config
